cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -428,4 +422,45 @@ private InstanceId createInstanceId(CConfiguration cConf) {
                                 Constants.INSTANCE_NAME, instanceName);
     return new InstanceId(instanceName);
   }
+
+  private void validateNoMappingUpdates(NamespaceConfig existingConfig,
+                                        NamespaceConfig updatedConfig) throws BadRequestException {
+    if (updatedConfig == null) {
+      // nothing to validate
+      return;
+    }
+
+    String updatedRootDirectory = updatedConfig.getRootDirectory();
```

Comment:
I think it's better to move all these to `NamespaceConfig` so that when there are new mapping added, it's harder to get missed. You can have something like a `boolean equalsMapping` method.